indexer/src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -95,138 +99,73 @@ class BackgroundExternalDatabaseSyncingService(
-                        //create new securable object for this column
-                        totalSynced += createSecurableColumnObjects(
-                                orgOwnerAclKeys,
-                                orgId,
-                                tableName,
-                                tableId,
-                                Optional.of(it),
-                                currentColumnIds
-                        )
-                    } else {
-                        currentColumnIds.add(columnId)
-                    }
-                }
+        val orgOwnerAclKeys = edms.getOrganizationOwners(orgId).map { it.aclKey }
 
-            }
+        val tablesToCols = edms.getColumnNamesByTableName(dbName).associate { (oid, tableName, schemaName, columnNames) ->
 
-        }
+            val table = getOrCreateTable(orgId, oid, tableName, schemaName, orgOwnerAclKeys)
+            val columnIds = syncTableColumns(table, orgOwnerAclKeys)
 
-        //check if tables have been deleted in the database
-        val missingTableIds = organizationExternalDatabaseTables
-                .filter { it.value.organizationId == orgId }
-                .keys - currentTableIds
-        if (missingTableIds.isNotEmpty()) {
-            edms.deleteOrganizationExternalDatabaseTableObjects(missingTableIds)
-            totalSynced += missingTableIds.size
+            table.id to columnIds
         }
 
-        //check if columns have been deleted in the database
-        val missingColumnIds = organizationExternalDatabaseColumns
-                .filter { it.value.organizationId == orgId }
-                .keys - currentColumnIds
+        removeNonexistentTablesAndColumnsForOrg(orgId, tablesToCols.keys, tablesToCols.flatMap { it.value }.toSet())
 
+    }
 
-        if (missingColumnIds.isNotEmpty()) {
-            val missingColumnsByTable = organizationExternalDatabaseColumns
-                    .getAll(missingColumnIds).entries
-                    .groupBy { it.value.tableId }
-                    .mapValues { it.value.map { entry -> entry.key!! }.toSet() }
+    private fun getOrCreateTable(orgId: UUID, oid: Int, tableName: String, schemaName: String, orgOwnerAclKeys: List<AclKey>): OrganizationExternalDatabaseTable {
+        val table = OrganizationExternalDatabaseTable(
+                Optional.empty(),
+                tableName,
+                tableName,
+                Optional.empty(),
+                orgId,
+                oid,
+                schemaName
+        )
 
-            edms.deleteOrganizationExternalDatabaseColumnObjects(missingColumnsByTable)
-            totalSynced += missingColumnIds.size
+        if (reservationService.isReserved(table.getUniqueName())) {
+            return organizationExternalDatabaseTables.getValue(reservationService.getId(table.getUniqueName()))
         }
 
-        return totalSynced
+        createSecurableTableObject(orgOwnerAclKeys, orgId, table)
+
+        return table
```

Comment:
`createSecurableTableObject` returns `newTableId`. Don't you need to update `table` with the new id prior to returning it?

Reply:
the id doesn't actually change